### PR TITLE
crate work component

### PR DESCRIPTION
**Pull request by devstm**
crate work component in profile page

### CODE REVIEW

File: client/src/Components/UserInfoCard/index.tsx [hunk 1]
```diff
@@ -1,84 +1,93 @@
...
+  userInfo: any;
 }
-function UserInfoCard({ userInfo }:UserInfoCardProps) {
+
+function UserInfoCard({ userInfo }: UserInfoCardProps) {
   const {
-    name, rate, phone, email, services, description, location,
+    user: {
+      // eslint-disable-next-line camelcase
+      first_name, last_name, services, image, phone, email, description, location,
```
Comment (LinaYahya) [inline client/src/Components/UserInfoCard/index.tsx:22]:
```suggestion
      first_name: firstName, last_name: lastName, services, image, phone, email, description, location,
```

File: client/src/Components/UserInfoCard/index.tsx [hunk 2]
```diff
@@ -1,84 +1,93 @@
...
-  };
+  userInfo: any;
 }
-function UserInfoCard({ userInfo }:UserInfoCardProps) {
+
+function UserInfoCard({ userInfo }: UserInfoCardProps) {
   const {
-    name, rate, phone, email, services, description, location,
+    user: {
+      // eslint-disable-next-line camelcase
```
Comment (LinaYahya) [inline client/src/Components/UserInfoCard/index.tsx:21]:
```suggestion
```
remove eslint rules disable

File: client/src/Components/UserInfoCard/index.tsx [hunk 3]
```diff
@@ -1,84 +1,93 @@
...
         <div className="image">
-          <Image width={100} src={image1} />
+          <Image width={100} src={image} />
         </div>
 
         <div className="content">
           <div className="name">
-            <h2>{name}</h2>
-            <Rate allowHalf defaultValue={rate} disabled className="rate" />
+            {/* eslint-disable-next-line camelcase */}
```
Comment (LinaYahya) [inline client/src/Components/UserInfoCard/index.tsx:35]:
```suggestion
```

File: client/src/Components/UserInfoCard/index.tsx [hunk 4]
```diff
@@ -1,84 +1,93 @@
...
+            <h2>{last_name}</h2>
+            <Rate allowHalf defaultValue={totalReviews} disabled className="rate" />
           </div>
-          <p>{location}</p>
+          <p>{location.city}</p>
           {t('servicesOffer')}
           <p className="services">
-            {services.map((service) => (
+            {services && services.map((service: any) => (
```
Comment (LinaYahya) [inline client/src/Components/UserInfoCard/index.tsx:44]:
```suggestion
            {services && services.map((service: string) => (
```

File: client/src/Pages/Profile.tsx [hunk 5]
```diff
@@ -1,26 +1,63 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import WorkCard from '../Components/WorkCard/WorkCard';
 import UserInfoCard from '../Components/UserInfoCard';
+// eslint-disable-next-line import/extensions
```
Comment (LinaYahya) [inline client/src/Pages/Profile.tsx:5]:
```suggestion
```
fix eslint errors

File: client/src/Pages/Profile.tsx [hunk 6]
```diff
@@ -1,26 +1,63 @@
...
+        const { data }: Request = await request('get', '/providers/2');
+        const { works }: { works: Works } = data;
+        const allWorks1: any = {
+          1: [...works],
+        };
+        setIsLoading(false);
+        setData(data);
+        setWorks(allWorks1);
+      } catch (error1: any) {
+        setError(error1?.data.msg);
```
Comment (LinaYahya) [inline client/src/Pages/Profile.tsx:29]: we need to setLoading to false even if error happened

File: client/src/Pages/Profile.tsx [hunk 7]
```diff
@@ -1,26 +1,63 @@
...
+      try {
+        const { data }: Request = await request('get', '/providers/2');
+        const { works }: { works: Works } = data;
+        const allWorks1: any = {
+          1: [...works],
+        };
+        setIsLoading(false);
+        setData(data);
+        setWorks(allWorks1);
+      } catch (error1: any) {
```
Comment (LinaYahya) [inline client/src/Pages/Profile.tsx:28]:
```suggestion
      } catch (err: Error) {
```

File: client/src/Pages/Profile.tsx [hunk 8]
```diff
@@ -1,26 +1,63 @@
...
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const { data }: Request = await request('get', '/providers/2');
+        const { works }: { works: Works } = data;
+        const allWorks1: any = {
+          1: [...works],
+        };
+        setIsLoading(false);
+        setData(data);
```
Comment (LinaYahya) [inline client/src/Pages/Profile.tsx:26]: no need to set works to data

File: client/src/utils/types/profile.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
+export type User = {
```
Comment (LinaYahya) [inline client/src/utils/types/profile.ts:1]: How Great!

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/Pages/Profile.tsx [hunk 10]
```diff
@@ -1,26 +1,66 @@
...
+          1: [...works],
+        };
+        const userInfoData = {
+          user: data.user,
+          reviews: data.reviews,
+          totalReviews: data.totalReviews,
+        };
+        setIsLoading(false);
+        setData(userInfoData);
+        setWorks(allWorks1);
```
Comment (LinaYahya) [inline client/src/Pages/Profile.tsx:29]:
```suggestion
        setWorks({
          1: [...works],
        });
```

File: client/src/Pages/Profile.tsx [hunk 11]
```diff
@@ -1,26 +1,66 @@
...
+          loading
+          {' '}
+          {error}
+        </h1>
+      ) : (
+        <>
+          <UserInfoCard userInfo={userData} />
+          <div className="work-card-container">
+            {allWorks[1].map((work: OnWork) => (
+              <WorkCard work={work} actions={act} isAuth={isAuth} />
```
Comment (LinaYahya) [inline client/src/Pages/Profile.tsx:57]: key is missing here

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by RandSohail — APPROVED** [review]

**Comment by devstm** [comment]
create work card component in profile page

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**